ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/clientlibs/site/js/fileinputwidget.js

```diff
@@ -19,7 +19,7 @@
  * file list, handling invalid file size, file name, file mime type functionality
  */
 if (typeof window.FileInputWidget === 'undefined') {
-    window.FileInputWidget = class extends FormView.FormFileInputWidget {
+    window.FileInputWidget = class extends FormView.FormFileInputWidgetBase {
```

Comment:
Why is there change in v1, what about the existing functionality diff between FormFileInputWidget and FormFileInputWidgetBase ? Wouldn't this cause backward compatibility issue

Reply:
Earlier when we only had v1 and v2, FormFileInputWidget was the base of both v1 and v2, and v2 (view) had extra functions for drag and drop. Now since V3 also has the same features as v2 with some extra functions so there would be a code duplicity between v2 and v3.
So we renamed the existing 'FormFileInputWidget' (which is now the base for v1 v2 and v3) to 'FormFileInputWidgetBase' and moved the common code between v2 and v3 into 'FormFileInputWidget'. So now if a customer only loads v1, then also it will work and similar for v2 and v3.

This is the reason we need to expose FormFileInputWidget and FormFileInputWidgetBase both as they are being used in the respective clientlibs.